src/geo/projection/globe.js

```diff
@@ -0,0 +1,499 @@
+            return interpolate(globePixelDistance, mercatorPixelDistance, t);
+        }
+        return globePixelDistance;
+    }
+};
+
+export function globeTileBounds(id: CanonicalTileID): Aabb {
+    const z = id.z;
+
+    const mn = -GLOBE_RADIUS;
+    const mx = GLOBE_RADIUS;
+
+    if (z === 0) {
+        return new Aabb([mn, mn, mn], [mx, mx, mx]);
+    } else if (z === 1) {
+        if (id.x === 0 && id.y === 0) {
+            return new Aabb([mn, mn, mn], [0, 0, mx]);
+        } else if (id.x === 1 && id.y === 0) {
+            return new Aabb([0, mn, mn], [mx, 0, mx]);
+        } else if (id.x === 0 && id.y === 1) {
+            return new Aabb([mn, 0, mn], [0, mx, mx]);
+        } else if (id.x === 1 && id.y === 1) {
+            return new Aabb([0, 0, mn], [mx, mx, mx]);
+        }
+    }
+
+    // After zoom 1 surface function is monotonic for all tile patches
+    // => it is enough to project corner points
+    const [min, max] = globeTileLatLngCorners(id);
+
+    const corners = [
+        latLngToECEF(min[0], min[1], GLOBE_RADIUS),
+        latLngToECEF(min[0], max[1], GLOBE_RADIUS),
+        latLngToECEF(max[0], min[1], GLOBE_RADIUS),
+        latLngToECEF(max[0], max[1], GLOBE_RADIUS)
+    ];
+
+    const bMin = [mx, mx, mx];
+    const bMax = [mn, mn, mn];
+
+    for (const p of corners) {
+        bMin[0] = Math.min(bMin[0], p[0]);
+        bMin[1] = Math.min(bMin[1], p[1]);
+        bMin[2] = Math.min(bMin[2], p[2]);
+
+        bMax[0] = Math.max(bMax[0], p[0]);
+        bMax[1] = Math.max(bMax[1], p[1]);
+        bMax[2] = Math.max(bMax[2], p[2]);
+    }
+
+    return new Aabb(bMin, bMax);
+}
+
+export function globeTileLatLngCorners(id: CanonicalTileID) {
+    const tileScale = Math.pow(2, id.z);
+    const left = id.x / tileScale;
+    const right = (id.x + 1) / tileScale;
+    const top = id.y / tileScale;
+    const bottom = (id.y + 1) / tileScale;
+
+    const latLngTL = [ latFromMercatorY(top), lngFromMercatorX(left) ];
+    const latLngBR = [ latFromMercatorY(bottom), lngFromMercatorX(right) ];
+
+    return [latLngTL, latLngBR];
+}
+
+export function latLngToECEF(lat: number, lng: number, radius: ?number): Array<number> {
+    lat = degToRad(lat);
+    lng = degToRad(lng);
+
+    if (!radius) {
+        radius = GLOBE_RADIUS;
+    }
+
+    // Convert lat & lng to spherical representation. Use zoom=0 as a reference
+    const sx = Math.cos(lat) * Math.sin(lng) * radius;
+    const sy = -Math.sin(lat) * radius;
+    const sz = Math.cos(lat) * Math.cos(lng) * radius;
+
+    return [sx, sy, sz];
+}
+
+function normalizeECEF(bounds: Aabb): Float64Array {
+    const m = mat4.identity(new Float64Array(16));
+
+    const maxExtInv = 1.0 / Math.max(...vec3.sub([], bounds.max, bounds.min));
+    const st = (1 << (NORMALIZATION_BIT_RANGE - 1)) - 1;
+
+    mat4.scale(m, m, [st, st, st]);
+    mat4.scale(m, m, [maxExtInv, maxExtInv, maxExtInv]);
+    mat4.translate(m, m, vec3.negate([], bounds.min));
+
+    return m;
```

Comment:
I don't think that's feasible without larger refactoring. gl-native uses mostly 16 bit integers for local tile coordinates [0, 8192] meaning that there's very little precision to spare. It would also increase memory usage quite a lot if these coordinates are upgraded to 32 bit values.